### PR TITLE
openssl version 3.0+ supported

### PR DESCRIPTION
**Pull request by WilburZjh**
Signed-off-by: Jinhang Zhang <Jinhang.Zhang@ibm.com>

### CODE REVIEW

**Comment by pshipton** [comment]
Setting as draft, see https://github.com/ibmruntimes/openj9-openjdk-jdk11/pull/499

**Comment by pshipton** [comment]
jenkins compile alinux64 jdk17

**Comment by pshipton** [comment]
There is a problem with the tracing, an error message occurs although it continues on properly.
```
Attempt to load OpenSSL /lib64/libcrypto.so.3
Error loading OpenSSL: Incompatible OpenSSL version found: OpenSSL 3.0.1 14 Dec 2021
Using OpenSSL version: OpenSSL 3.0.1 14 Dec 2021 (/lib64)
MessageDigest load - using Native crypto library.
NativeChaCha20Cipher load - using Native crypto library.
NativeGaloisCounterMode load - using Native crypto library.
Crypto test COMPLETED
```

File: closed/src/java.base/share/native/libjncrypto/NativeCrypto.c [hunk 1]
```diff
@@ -272,6 +281,11 @@ JNIEXPORT jint JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_loadCrypto
         ossl_ver = 1;
     }
 
+    if (trace) {
```
Comment (pshipton) [inline closed/src/java.base/share/native/libjncrypto/NativeCrypto.c:284]: These lines shouldn't be copied here, they need to be deleted.

Comment (pshipton) [reply closed/src/java.base/share/native/libjncrypto/NativeCrypto.c:284]: Actually it just seems the wrong string, it should be `"Supported OpenSSL version: %s\n"`

Comment (pshipton) [reply closed/src/java.base/share/native/libjncrypto/NativeCrypto.c:284]: Except on jdk17 there is already a trace message lower down on lines 432 and 435 which doesn't exist in the jdk11 code, so this one isn't needed.

Comment (WilburZjh) [reply closed/src/java.base/share/native/libjncrypto/NativeCrypto.c:284]: Deleted.

**Comment by pshipton** [comment]
It's not worth re-testing due to the change. We'll wait for Keith's review.

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**